libbpfgo.h

```diff
@@ -22,16 +22,25 @@ int libbpf_print_fn(enum libbpf_print_level level, // libbpf print level
                     const char *format,            // format used for the msg
                     va_list args) {                // args used by format
 
-  int ret = 0;
+  int ret;
+  size_t len;
   char *out;
   va_list check;
 
-  out = (char *)calloc(1, 300);
+  va_copy(check, args);
+  ret = vsnprintf(NULL, 0, format, check); // get output length
+  va_end(check);
```

Comment:
I didn't use va_args much in the past, but do we need to call va_copy and va_end twice in this function (here and below when using vsnprintf)?

Reply:
stdarg man stands:

```
...
If ap is passed to a function that uses va_arg(ap,type), then the value of ap is undefined after the return of that function.
```

So, being cautious, as the first call of vsnprintf iterates all args, I assume it's best to reset it.

Reply:
vsnprintf man:

```
The functions vprintf(), vfprintf(), vdprintf(), vsprintf(), vsnprintf() are equivalent to the functions printf(), fprintf(), dprintf(), sprintf(), snprintf(), respectively, except that they are called with a va_list instead of a variable number of arguments. These functions do not call the va_end macro. Because they invoke the va_arg macro, the value of ap is undefined after the call. See stdarg(3).
```

Reply:
What I thought is that maybe you can call va_copy once (where you call it first now), and only leave the last va_end.
But sice I didn't read the man of va_args and I believe you did, you can ignore this comment if it is not relevant.